pkg/cmd/get/get_preview_test.go

```diff
@@ -0,0 +1,160 @@
+		if err.Error() != want {
+			t.Errorf("CurrentPreviewUrl() failed with wrong error got `%s' want `%s'", err.Error(), want)
+		}
+	} else {
+		t.Errorf("CurrentPreviewUrl() didn't fail, expected error: `%s'", want)
+	}
+}
+
+func TestCurrentPreviewUrlWithoutPreviews(t *testing.T) {
+	os.Setenv("REPO_OWNER", "jx-testing")
+	os.Setenv("REPO_NAME", "jx-testing")
+	defer os.Unsetenv("REPO_OWNER")
+	defer os.Unsetenv("REPO_NAME")
+
+	commonOpts := &opts.CommonOptions{}
+	commonOpts.SetDevNamespace("jx-testing")
+
+	factory := clientmocks.NewMockFactory()
+	user := jenkinsv1.User{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "jx",
+			Name:      "test-user",
+		},
+		Spec: jenkinsv1.UserDetails{
+			Name:     "Test",
+			Email:    "test@test.com",
+			Accounts: make([]jenkinsv1.AccountReference, 0),
+		},
+	}
+
+	testhelpers.ConfigureTestOptionsWithResources(commonOpts,
+		[]runtime.Object{},
+		[]runtime.Object{&user},
+		&gits.GitFake{CurrentBranch: "job"},
+		&gits.FakeProvider{},
+		helm_test.NewMockHelmer(),
+		resources_test.NewMockInstaller(),
+	)
+
+	commonOpts.SetFactory(factory)
+
+	options := &get.GetPreviewOptions{
+		GetEnvOptions: get.GetEnvOptions{
+			GetOptions: get.GetOptions{
+				CommonOptions: commonOpts,
+			},
+		},
+	}
+
+	want := "No Preview for name: jx-testing-jx-testing-job"
+
+	if _, err := options.CurrentPreviewUrl(); err != nil {
+		if err.Error() != want {
+			t.Errorf("CurrentPreviewUrl() failed with wrong error `got' %s want `%s'", err.Error(), want)
+		}
+	} else {
+		t.Errorf("CurrentPreviewUrl() didn't fail, expected error: `%s'", want)
+	}
+}
+
+func TestCurrentPreviewUrlWithPreviews(t *testing.T) {
+	os.Setenv("REPO_OWNER", "jx-testing")
+	os.Setenv("REPO_NAME", "jx-testing")
+	defer os.Unsetenv("REPO_OWNER")
+	defer os.Unsetenv("REPO_NAME")
+
+	commonOpts := &opts.CommonOptions{}
+	commonOpts.SetDevNamespace("jx")
+
+	factory := clientmocks.NewMockFactory()
+	user := jenkinsv1.User{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "jx",
+			Name:      "test-user",
+		},
+		Spec: jenkinsv1.UserDetails{
+			Name:     "Test",
+			Email:    "test@test.com",
+			Accounts: make([]jenkinsv1.AccountReference, 0),
+		},
+	}
+
+	var devEnv *jenkinsv1.Environment
+	devEnv = kube.NewPreviewEnvironment("jx-testing-jx-testing-job")
+	devEnv.Spec.PreviewGitSpec.ApplicationURL = "http://example.com"
+
+	testhelpers.ConfigureTestOptionsWithResources(commonOpts,
+		[]runtime.Object{},
+		[]runtime.Object{&user, devEnv},
+		&gits.GitFake{CurrentBranch: "job"},
+		&gits.FakeProvider{},
+		helm_test.NewMockHelmer(),
+		resources_test.NewMockInstaller(),
+	)
+
+	commonOpts.SetFactory(factory)
+
+	options := &get.GetPreviewOptions{
+		GetEnvOptions: get.GetEnvOptions{
+			GetOptions: get.GetOptions{
+				CommonOptions: commonOpts,
+			},
+		},
+	}
+
+	want := "http://example.com"
+
+	if got, err := options.CurrentPreviewUrl(); err != nil {
+		t.Errorf("CurrentPreviewUrl() didn't expect error but errored with: `%s'", err)
```

Comment:
If I'm understanding the test failure correctly, I think this also needs `JOB_NAME` to be overridden like in the above test.